### DISCUSSION TITLE
Keep order in the filename?

### DISCUSSION THREAD

**Discussion by firedev** [bridgetownrb/bridgetown#825] [Q&A]
Hi,

Is there a way to strip the leading number from the filename when publishing so I could keep thoughts in order? I am using the same system you have to documentation site. There is `order:` in frontmatter, is there a way to keep the files sorted the same way? Some `permalink` magic?

**Comment by KonnorRogers** [comment]
Assuming you mean something like: `01-foo.md`

I _think_ you could modify the frontmatter in a `generator` hook. I haven't tested it for permalinks, but I did it for some other frontmatter defaults.

https://www.bridgetownrb.com/docs/plugins/generators

Something like this:

```rb
# plugins/builders/permalink_modifier.rb
class Builders::PermalinkModifier < SiteBuilder
  def build
    generator do
       # Rename "thoughts" to whatever your collection name is.
       resources = site.collections.thoughts.resources
      
       resources.each do |resource|
         thought_name = File.basename(resource.relative_path.basename, ".md").to_s
         stripped_thought = thought_name.split(/^\d+-/).join("")

         # Modify the frontmatter data to have the right permalink
         resource.data.merge!({
           permalink: "/thoughts/#{stripped_thought}"
         })
       end
    end
  end
end
```

Perhaps a build hook may make more sense.

https://www.bridgetownrb.com/docs/plugins/hooks

Happy to defer to @jaredcwhite here if this isn't the right way!
